cpp/src/arrow/compute/kernels/hash_aggregate_test.cc

```diff
@@ -2087,7 +2363,7 @@ TEST(GroupBy, AnyAndAll) {
     auto keep_nulls_min_count =
         std::make_shared<ScalarAggregateOptions>(/*skip_nulls=*/false, /*min_count=*/3);
     ASSERT_OK_AND_ASSIGN(Datum aggregated_and_grouped,
-                         RunGroupBy(
+                         AltGroupBy(
```

Comment:
Why change the function name here and what does "Alt" in "AltGroupBy" mean?

Reply:
The [`AltGroupBy` function](https://github.com/apache/arrow/pull/34311/files#diff-ec1e8907dfc3e636abecd3307879f331421e4f28f75e504f147826cebf04d7e0R1116-R1123) chooses an alternative for the group-by invocation. There are 2 alternatives: `RunGroupByImpl` and `RunSegmentedGroupByImpl`.